frontend/src/app/main/ui/shapes/export.cljs

```diff
@@ -286,32 +287,41 @@
     (mf/html
      [:> "penpot:fills" #js {}
       (for [[index fill] (d/enumerate fills)]
-        [:> "penpot:fill"
-         #js {:penpot:fill-color          (if (some? (:fill-color-gradient fill))
-                                              (str/format "url(#%s)" (str "fill-color-gradient_" (mf/use-ctx muc/render-id) "_" index))
+        (let [fill-image-id (dm/str "fill-image-" (mf/use-ctx muc/render-id) "-" index)]
+          [:> "penpot:fill"
+           #js {:penpot:fill-color          (cond
+                                              (some? (:fill-color-gradient fill))
+                                              (str/format "url(#%s)" (str "fill-color-gradient-" (mf/use-ctx muc/render-id) "-" index))
+
+                                              :else
                                               (d/name (:fill-color fill)))
-              :penpot:fill-color-ref-file (d/name (:fill-color-ref-file fill))
-              :penpot:fill-color-ref-id   (d/name (:fill-color-ref-id fill))
-              :penpot:fill-opacity        (d/name (:fill-opacity fill))}])])))
+                :penpot:fill-image-id       (when (:fill-image fill) fill-image-id)
+                :penpot:fill-color-ref-file (d/name (:fill-color-ref-file fill))
+                :penpot:fill-color-ref-id   (d/name (:fill-color-ref-id fill))
+                :penpot:fill-opacity        (d/name (:fill-opacity fill))}]))])))
 
 (defn- export-strokes-data [{:keys [strokes]}]
   (when-let [strokes (seq strokes)]
     (mf/html
      [:> "penpot:strokes" #js {}
       (for [[index stroke] (d/enumerate strokes)]
-        [:> "penpot:stroke"
-         #js {:penpot:stroke-color          (if (some? (:stroke-color-gradient stroke))
-                                              (str/format "url(#%s)" (str "stroke-color-gradient_" (mf/use-ctx muc/render-id) "_" index))
-                                              (d/name (:stroke-color stroke)))
-              :penpot:stroke-color-ref-file (d/name (:stroke-color-ref-file stroke))
-              :penpot:stroke-color-ref-id   (d/name (:stroke-color-ref-id stroke))
-              :penpot:stroke-opacity        (d/name (:stroke-opacity stroke))
-              :penpot:stroke-style          (d/name (:stroke-style stroke))
-              :penpot:stroke-width          (d/name (:stroke-width stroke))
-              :penpot:stroke-alignment      (d/name (:stroke-alignment stroke))
-              :penpot:stroke-cap-start      (d/name (:stroke-cap-start stroke))
-              :penpot:stroke-cap-end        (d/name (:stroke-cap-end stroke))}])])))
-
+        (let [stroke-image-id (dm/str "stroke-image-" (mf/use-ctx muc/render-id) "-" index)]
```

Comment:
use-ctx incorreclty used, should be declared on the toplevel let of the component